scripting/weaponmodels/entitydata.sp

```diff
@@ -118,72 +128,147 @@ void WeaponModels_EntityDataInit()
 
-	InitSendPropOffset(g_iOffset_ViewModelOwner, "CBaseViewModel", "m_hOwner");
-	InitSendPropOffset(g_iOffset_ViewModelWeapon, "CBaseViewModel", "m_hWeapon");
-	InitSendPropOffset(g_iOffset_ViewModelSequence, "CBaseViewModel", "m_nSequence");
-	InitSendPropOffset(g_iOffset_ViewModelPlaybackRate, "CBaseViewModel", "m_flPlaybackRate");
-	InitSendPropOffset(g_iOffset_ViewModelIndex, "CBaseViewModel", "m_nViewModelIndex");
+	InitSendPropOffset(g_iOffset_ViewModelOwner, "CBaseViewModel", "m_hOwner", true);
+	InitSendPropOffset(g_iOffset_ViewModelWeapon, "CBaseViewModel", "m_hWeapon", true);
+	InitSendPropOffset(g_iOffset_ViewModelSequence, "CBaseViewModel", "m_nSequence", true);
+	InitSendPropOffset(g_iOffset_ViewModelPlaybackRate, "CBaseViewModel", "m_flPlaybackRate", true);
+	InitSendPropOffset(g_iOffset_ViewModelIndex, "CBaseViewModel", "m_nViewModelIndex", true);
 	
 	InitSendPropOffset(g_iOffset_ViewModelIgnoreOffsAcc, "CBaseViewModel", "m_bShouldIgnoreOffsetAndAccuracy", false);
 	InitSendPropOffset(g_iOffset_EconItemDefinitionIndex, "CEconEntity", "m_iItemDefinitionIndex", false);
 
 	int lightingOriginOffset;
-	InitSendPropOffset(lightingOriginOffset, "CBaseAnimating", "m_hLightingOrigin");
+	InitSendPropOffset(lightingOriginOffset, "CBaseAnimating", "m_hLightingOrigin", true);
 
 	// StudioHdr offset in gameconf is only relative to the offset of m_hLightingOrigin, in order to make the offset more resilient to game updates
 	g_iOffset_Animating_StudioHdr += lightingOriginOffset;
 }
 
-int GetPlayerViewModel(int client, int index)
+public int GetPlayerViewModel(int client, int index)
 {
+	#if defined DEBUG
+	PrintToServer("GetPlayerViewModel");
+	#endif
 	return GetEntDataEnt2(client, g_iOffset_PlayerViewModel + (index * 4));
 }
 
-void SetPlayerViewModel(int client, int index, int viewModel)
+public void SetPlayerViewModel(int client, int index, int viewModel)
 {
+	#if defined DEBUG
+	PrintToServer("SetPlayerViewModel");
+	#endif
 	SetEntDataEnt2(client, g_iOffset_PlayerViewModel + (index * 4), viewModel, true);
 }
 
-void SetEntityVisibility(int entity, bool show)
-{
-	int flags = GetEntData(entity, g_iOffset_EntityEffects);
+public void SetEntityVisibility(int entity, bool show){
+
+	#if defined DEBUG
+	PrintToServer("Seting EntityVisibility for entity: %d", entity);
+	#endif
 
+	int flags = GetEntData(entity, g_iOffset_EntityEffects);
 	SetEntData(entity, g_iOffset_EntityEffects, show ? flags & ~EF_NODRAW : flags | EF_NODRAW, _, true);
 }
 
+public void NextFrameSetVisibility(DataPack dataPackHandle){
+
+	ResetPack(dataPackHandle);
+ 	int framestoskip = dataPackHandle.ReadCell();
+ 	if ( framestoskip > 0 )		//recursive frame skip	
+	{
+		ResetPack(dataPackHandle);
+ 		dataPackHandle.WriteCell( framestoskip-1 );
+		
+ 		RequestFrame(NextFrameSetVisibility, dataPackHandle);
+ 		return;
+	}
+	int show = dataPackHandle.ReadCell();
+	int entity = dataPackHandle.ReadCell();
```

Comment:
you should always use entity reference rather than index when doing stuff to an entity after a delay. entity ref is unique, index is not. There is a change that the entity has been deleted during the delay and the index got reused for a completely different entity that has spawned. See EntIndexToEntRef() and EntRefToEntIndex()

Reply:
I see, will keep that mind!